Add RPC GET chains/main/blocks/<block_hash>/votes/listings

adonagy opened the pull request:
This PR adds the RPC chains/main/blocks/<block_hash>/votes/listings. The RPC returns the delegates, that are eligible to vote, with their vote's weights in rolls

Kyras commented on rpc/src/encoding/conversions.rs:
```diff
@@ -76,6 +76,27 @@ pub fn public_key_to_contract_id(pk: Vec<u8>) -> Result<String, failure::Error>
+        };
+        Ok(contract_id)
+    } else {
+        bail!("Invalid hash")
```
It would be better to define custom error through `failure::Fail`, rather than bail. Not crucial thou.

Kyras commented on rpc/src/server/service.rs:
```diff
@@ -668,6 +670,41 @@ pub(crate) fn get_rolls_owner_current_from_context(level: &str, list: ContextLis
+
+    // filter out the listings data
+    let listings_data: ContextMap = ctxt.unwrap().into_iter()
+        .filter(|(k, _)| k.contains(&"data/votes/listings/"))
```
If it is possible, use `starts_wtih` instead of `contains`. (Don't remember if it is prefix or not).
EDIT: Just notice in further comment it should be an prefix.

adonagy commented:
@Kyras I've added the required changes and also reworked the bail usage and created an error enum. Please check and let me know whether it is ok :)

Also added missing unit tests in the conversions
